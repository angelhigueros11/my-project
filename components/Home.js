import { StatusBar } from 'expo-status-bar';
import React,{useState,useEffect} from 'react';
import { StyleSheet, Text, View,SafeAreaView,Image,TouchableOpacity,Platform,Alert,ScrollView} from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { FontAwesome } from '@expo/vector-icons'; 
import FormView from './FormView';
import Camara from './Camara.js';
import CreateProfile from './CreateProfile';
import firebase from './firebase2.js';


function getForms(email){
  let lista = [];
  
  return lista;
}



export default function Home({navigation,route}) {

  const {name,email,photoUrl} = route.params;
  const AvatarImg = (photoUrl)? {uri:photoUrl}: require('../assets/src/icons/avatar.png');
  const AvatarEmail = (email)? email:"Aun no se registra un correo";
  const [listaFormularios,setListaFormularios] = useState([]);
  
  
  useEffect(() => {
    
    firebase.db.collection(email).onSnapshot(querySnapshot => {
      let lista = [];
      querySnapshot.docs.forEach(doc => {
        const {titulo,nombre,apellido,edad,dpi,birth,smoker,med,pregnant,date} = doc.data();
        const data = {
          name: nombre,
          lastName: apellido,
          age: edad,
          document: dpi,
          birth: birth,
          smoker:smoker,
          med:med,
          pregnant: pregnant,
        }
        lista.push(
          <FormView name={titulo} date={date} data={data}/>
        );
      })
      setListaFormularios(lista);
    })
  },[])

  // useEffect(() =>{
  //   setListaFormularios(getForms(email));
  // },[])

  return (
    <SafeAreaView style={styles.container}>
      {/* User Info Container.*/}
      <View style={styles.userInfoContainer}>
        <View style={{flexDirection:'row',flex:0.9, backgroundColor:'#143590',alignItems: 'center'}}>
          <View style={styles.userImageContainer}>
            <Image source={AvatarImg} style={styles.userImage}/>
          </View>
          <View style={styles.userTextInfoContainer}>
            <Text style={styles.userNameText}>{name}</Text>
            <Text style={styles.userInfoText}>{AvatarEmail}</Text>
          </View>
        </View>
        <TouchableOpacity style={styles.settingsIcon} onPress={()=>{
          navigation.navigate('EditProfile',{name,email,photoUrl});
          
          
          
          }}>
          <Ionicons name="md-settings-sharp" size={28} color="#fff" />
        </TouchableOpacity>
      </View>

      {/* MAIN AREA   */}
      <View style={styles.mainAreaContainer}>
        <View style={styles.firstRowContainer}>
          <TouchableOpacity style={styles.searchIconContainer}>
            <FontAwesome name="search" size={28} color="#fff" />
          </TouchableOpacity>
          <Text style={styles.formsTitle}>Formularios</Text>
        </View>

        {/* Form view */}
        <View style={ styles.secondRowContainer }>

        <ScrollView  horizontal={true} >

          {/* <FormView name="Covid" date="02/02/2022" />
          <FormView name="Vacuna" date="12/03/2019" />
          <FormView name="General" date="11/04/2022" />
          <FormView name="Extra" date="12/12/2021" /> */}
          {listaFormularios}

        </ScrollView>
        </View>



        <TouchableOpacity style={styles.scanButton} onPress={()=>{navigation.navigate("Camara")}}>
          <Ionicons name="md-scan" size={24} color="white" />
          <Text style={styles.scanButtonText}>ESCANEAR</Text>
        </TouchableOpacity>
        <TouchableOpacity style={styles.scanButton2} onPress={()=>{navigation.navigate("CreateForm",{name,email,photoUrl})}}>
          <Text style={styles.scanButtonText2}>Crear Formulario</Text>
        </TouchableOpacity>
      </View>

     
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#143590',
    alignItems: 'center',
    justifyContent: 'flex-start',
    paddingTop: (Platform.OS === 'android')? 30:0,
  },
  userInfoContainer:{
    flex: 0.15,
    backgroundColor: '#143590',
    width: '90%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  userImageContainer: {
    left:10
  },
  userImage:{
    width: 50,
    height:50,
    borderRadius:150,
  },
  userTextInfoContainer:{
    backgroundColor: '#143590',
    height:'100%',
    justifyContent: 'flex-start',
    marginLeft:20,
    maxWidth:200
  },
  userNameText:{
    fontWeight: 'bold',
    fontSize:16,
    color: '#fff',
  },
  userInfoText:{
    fontSize:13,
    color:'#bbbfc3'
  },
  settingsIcon:{
    backgroundColor: '#143590',
    right:20,
   
  },
  mainAreaContainer:{
    flex:0.8,
    backgroundColor: '#143590',
    width:'90%',
  },
  firstRowContainer:{
    flexDirection: 'row-reverse',
    flex:0.3,
    backgroundColor: '#143590',
    alignItems: 'center',
    justifyContent: 'space-between',
    marginBottom:10
    
  },
  formsTitle:{
    fontSize:20,
    fontWeight: 'bold',
    marginLeft:20,
    color: '#fff',
    

  },searchIconContainer:{
    left:20,
    backgroundColor: '#143590'
  },
  secondRowContainer:{
    marginTop:10,
    height: 270, 
    backgroundColor: '#143590',
    marginBottom:20

   
    
    
  },scanButton:{
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#0657a2',
    borderRadius:8,
    height: 50,
    marginTop:10,
    bottom:10,
    width:'80%',
    alignSelf: 'center',
    
  },scanButton2:{
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#143590',
    borderRadius:8,
    height: 50,
    marginTop:10,
    bottom:10,
    width:'80%',
    alignSelf: 'center',
    color: 'white',
    
  },scanButtonText:{
    color: 'white',
    fontSize:15,
    marginLeft:10
  },scanButtonText2:{
    color: 'white',
    fontSize:15,
    marginLeft:10
  },
  addFloatingButton:{
    position:'absolute',
    backgroundColor:'#f15a24',
    top:'78%',
    left:'70%',
    height:60,
    width:60,
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius:150
  },
  addFloatingButtonText:{
    color: 'white',
    fontSize:40,
    bottom:5,
    
  }



});
