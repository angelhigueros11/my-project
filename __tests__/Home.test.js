import React from 'react';
import renderer from 'react-test-renderer';
import './Mocks/addEventListener.mock';
import Home from '../components/Home';

describe('Pruebas en <Home />', () => {
    it('Renders correctly', () => {
        const name = 'Nombre';
        const email = 'Email';
        const photoUrl = 'Photo';
        const route = { params: {name,email,photoUrl}};
        const tree = renderer.create(<Home route={route} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly withoutParams', () => {

        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();        
    });
});