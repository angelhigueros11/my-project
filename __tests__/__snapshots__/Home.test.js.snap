// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CreateProfile /> Renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#143590",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": 0,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#143590",
        "flex": 0.15,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "90%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#143590",
          "flex": 0.9,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "left": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "borderRadius": 150,
              "height": 50,
              "width": 50,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#143590",
            "height": "100%",
            "justifyContent": "flex-start",
            "marginLeft": 20,
            "maxWidth": 200,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Nombre
        </Text>
        <Text
          style={
            Object {
              "color": "#bbbfc3",
              "fontSize": 13,
            }
          }
        >
          Email
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#143590",
          "opacity": 1,
          "right": 20,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#143590",
        "flex": 0.8,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 20,
        }
      }
    >
      Formularios
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#143590",
          "height": 270,
          "marginBottom": 20,
          "marginTop": 10,
        }
      }
    >
      <RCTScrollView
        horizontal={true}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "alignSelf": "center",
                "backgroundColor": "#143590",
                "borderRadius": 15,
                "bottom": 20,
                "height": 190,
                "justifyContent": "center",
                "marginLeft": 20,
                "marginTop": 20,
                "overflow": "hidden",
                "width": 250,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#fff",
                  "fontSize": 25,
                }
              }
            >
              Error
            </Text>
          </View>
        </View>
      </RCTScrollView>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#0657a2",
          "borderRadius": 8,
          "bottom": 10,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginLeft": 10,
          }
        }
      >
        ESCANEAR
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#143590",
          "borderRadius": 8,
          "bottom": 10,
          "color": "white",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginTop": 10,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginLeft": 10,
          }
        }
      >
        Crear Formulario
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#143590",
          "borderRadius": 8,
          "bottom": 10,
          "color": "white",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "marginTop": 40,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 15,
            "marginLeft": 10,
          }
        }
      >
        Terminos y condiciones
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`Pruebas en <CreateProfile /> Renders correctly withoutParams 1`] = `
<View>
  <Text>
    ERROR
  </Text>
</View>
`;
